client/cli/src/params/node_key_params.rs

```diff
@@ -124,16 +121,36 @@ fn parse_ed25519_secret(hex: &str) -> error::Result<sc_network::config::Ed25519S
 	H256::from_str(&hex)
 		.map_err(invalid_node_key)
 		.and_then(|bytes| {
-			sc_network::config::identity::ed25519::SecretKey::from_bytes(bytes)
+			ed25519::SecretKey::from_bytes(bytes)
 				.map(sc_network::config::Secret::Input)
 				.map_err(invalid_node_key)
 		})
 }
 
+/// Open a file and try to parse it as hex or raw bytes Ed25519 secret key into a `sc_network::Secret`.
+fn open_ed25519_secret(file: &Path) -> error::Result<sc_network::config::Ed25519Secret> {
+	let mut file = File::open(file)
+		.map_err(|e| format!("Failed to open ed25519 secret: {:?}", e))?;
+
+	let mut content = Vec::new();
+	file.read_to_end(&mut content).map_err(|e| format!("Failed to read ed25519 secret: {:?}", e))?;
+
+	let secret_key = match String::from_utf8(content.clone())
+		.ok()
+		.and_then(|s| H256::from_str(&s).ok())
```

Comment:
Can we please check `content.len()` to be either 32 or 64 instead? I don't really trust `H256::from_str` to be future-proof.

Reply:
It is under our "control" and the docs already say that they do exactly these checks: https://docs.rs/fixed-hash/0.6.1/src/fixed_hash/hash.rs.html#595

Reply:
> It is under our "control"

But we're not an omniscient collective brain.
I totally imagine in a scenario where in a year from now someone changes `H256` to automatically add 0s if the value is not long enough, or something like that, and suddenly private keys that only contain valid hexadecimal characters will parse incorrectly.